pyomo/core/expr/taylor_series.py

```diff
@@ -0,0 +1,22 @@
+from pyomo.core.expr.current import identify_variables, value
+from pyomo.core.expr.calculus.derivatives import differentiate
+
+
+def get_taylor_series(expr, diff_mode=differentiate.Modes.reverse_numeric):
```

Comment:
Why `get_taylor_series`?  We are not really retrieving a Taylor series (I generally think of `get_` as methods on a class).  What do you think about something like `Taylor_series_expansion()`?

Also, this method is only for first-order Taylor series... would you consider adding an `order=n` option for getting higher order expansions?  (Of course, the default order would be 1, and the mode would be restricted to symbolic derivatives for higher orders...)

That would make a user's experience something like:

```python
model.c = Constraint(expr=model.y == Taylor_series_expansion(model.x**3, order=2)
```

Reply:
Yes. Good suggestion.